bolt-contracts/src/contracts/BoltChallenger.sol

```diff
@@ -22,26 +30,41 @@ contract BoltChallenger is IBoltChallenger {
 
     // ========= STORAGE =========
 
+    /// @notice Bolt Parameters contract.
+    IBoltParameters public parameters;
+
     /// @notice The set of existing unique challenge IDs.
     EnumerableSet.Bytes32Set internal challengeIDs;
```

Comment:
note: when counting storage layout, each variable should be counted for the amount of storage it uses.
In this case, `EnumerableSet.Bytes32Set` is defined as `Set` under the hood:

```solidity
    struct Set {
        bytes32[] _values;
        mapping(bytes32 value => uint256) _positions;
    }
```

this occupies 2 slots: 1 for the array (pointer to its mem address) and 1 for the mapping.
reference: https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#mappings-and-dynamic-arrays

so TL;DR: storage layout marker would be 4 instead of 3 slots here. I didn't check other contracts but this is a good check to make everywhere

Reply:
Good catch

Reply:
Validated all storage layouts with `forge inspect BoltParameters storage-layout --pretty`

Added instructions on how to do it in the contract natspec documentation. The OpenZeppelin foundry upgrades toolkit can also validate it every time we do a new deployment.